packages/design-system/src/components/input/useNumericInput.tsx

```diff
@@ -58,20 +60,36 @@ const useNumericInput = ({
     (ev: unknown) => {
       let newValue = parseInput(oldValue.current, options);
 
-      if (!revertToOriginal.current) {
+      if (!revertToOriginal.current && newValue !== null) {
         const parsedValue = parseInput(currentValue, options);
 
         if (parsedValue !== null) {
           newValue = parsedValue;
         }
+      } else if (currentValue !== null) {
+        newValue = parseInput(currentValue, options);
       }
 
       revertToOriginal.current = false;
       if (newValue !== null) {
+        // Set newly updated value.
         setCurrentValue(newValue);
         if (newValue !== oldValue.current) {
           onChange(ev, newValue);
         }
+      } else if (min !== undefined) {
+        // When new value is null and min is defined, set it to min value.
+        setCurrentValue(min);
+        onChange(ev, min);
+      } else if (!allowEmpty) {
+        // When above condition do not meet and empty values are not allowed, set value to `0`.
+        // TODO (@AnuragVasanwala): Improve this logic by considering min and max boundary.
+        setCurrentValue('0');
+        onChange(ev, '0');
+      } else {
+        // When none of the above condition met, set empty value.
+        setCurrentValue('');
+        onChange(ev, '');
       }
     },
     [currentValue, onChange, options]
```

Comment:
```suggestion
    [currentValue, onChange, options, allowEmpty, min]
```

To fix the eslint error

Reply:
Fixed in the latest commit: https://github.com/GoogleForCreators/web-stories-wp/pull/13339/commits/ea68d356ef699d864f2a6ee752341e97ab4b9a77